### ISSUE TITLE
Editing update requests result in incomplete changes

### ISSUE THREAD

**Issue by ellert** [fedora-infra/bodhi#679]
Hi

I tried to edit the update https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2015-70f2d35019

This update has three builds:

- dpm-dsi-1.9.5-9.el7
- globus-xio-udt-driver-1.19-1.el7
- globus-gridftp-server-8.9-1.el7

I tried to add one new build:

- globus-common-16.0-1.el7

And update two of the existing ones:

- dpm-dsi-1.9.5-9.el7 → dpm-dsi-1.9.5-10.el7
- globus-gridftp-server-8.9-1.el7 → globus-gridftp-server-9.0-2.el7

When submitting the changes the spinning arrow on the green Edit button appeared for a while, and the nothing happened. The update as it appears in bodhi still contains the original three builds.

However in koji, the two builds that were going to be replaced were untagged, but none of the three new ones were tagged. The one that should remain unchanged did not change.
```
$ koji list-tag-history --build dpm-dsi-1.9.5-9.el7
Sun Oct 25 15:13:37 2015: dpm-dsi-1.9.5-9.el7 tagged into epel7-testing-candidate by ellert
Sun Oct 25 15:24:30 2015: dpm-dsi-1.9.5-9.el7 tagged into epel7-testing-pending by bodhi
Mon Oct 26 20:32:06 2015: dpm-dsi-1.9.5-9.el7 untagged from epel7-testing-candidate by bodhi
Mon Oct 26 20:32:06 2015: dpm-dsi-1.9.5-9.el7 tagged into epel7-testing by bodhi
Mon Oct 26 20:32:37 2015: dpm-dsi-1.9.5-9.el7 untagged from epel7-testing-pending by bodhi
Tue Oct 27 13:37:35 2015: dpm-dsi-1.9.5-9.el7 untagged from epel7-testing by bodhi
Tue Oct 27 13:37:35 2015: dpm-dsi-1.9.5-9.el7 tagged into epel7-testing-candidate by bodhi [still active]

$ koji list-tag-history --build dpm-dsi-1.9.5-10.el7
Tue Oct 27 11:53:15 2015: dpm-dsi-1.9.5-10.el7 tagged into epel7-testing-candidate by ellert [still active]

$ koji list-tag-history --build globus-xio-udt-driver-1.19-1.el7
Fri Oct 23 18:33:03 2015: globus-xio-udt-driver-1.19-1.el7 tagged into epel7-testing-candidate by ellert
Sat Oct 24 05:07:02 2015: globus-xio-udt-driver-1.19-1.el7 tagged into epel7-testing-pending by bodhi
Mon Oct 26 20:31:58 2015: globus-xio-udt-driver-1.19-1.el7 untagged from epel7-testing-candidate by bodhi
Mon Oct 26 20:31:58 2015: globus-xio-udt-driver-1.19-1.el7 tagged into epel7-testing by bodhi [still active]
Mon Oct 26 20:32:37 2015: globus-xio-udt-driver-1.19-1.el7 untagged from epel7-testing-pending by bodhi

$ koji list-tag-history --build globus-gridftp-server-8.9-1.el7
Fri Oct 23 21:12:01 2015: globus-gridftp-server-8.9-1.el7 tagged into epel7-testing-candidate by ellert
Sat Oct 24 05:07:01 2015: globus-gridftp-server-8.9-1.el7 tagged into epel7-testing-pending by bodhi
Sun Oct 25 12:08:43 2015: globus-gridftp-server-8.9-1.el7 tagged into epel7-override by bodhi
Mon Oct 26 20:31:57 2015: globus-gridftp-server-8.9-1.el7 untagged from epel7-testing-candidate by bodhi
Mon Oct 26 20:31:57 2015: globus-gridftp-server-8.9-1.el7 tagged into epel7-testing by bodhi
Mon Oct 26 20:32:37 2015: globus-gridftp-server-8.9-1.el7 untagged from epel7-testing-pending by bodhi
Tue Oct 27 10:58:29 2015: globus-gridftp-server-8.9-1.el7 untagged from epel7-override by bodhi
Tue Oct 27 13:37:38 2015: globus-gridftp-server-8.9-1.el7 untagged from epel7-testing by bodhi
Tue Oct 27 13:37:38 2015: globus-gridftp-server-8.9-1.el7 tagged into epel7-testing-candidate by bodhi [still active]

$ koji list-tag-history --build globus-gridftp-server-9.0-2.el7
Tue Oct 27 13:11:44 2015: globus-gridftp-server-9.0-2.el7 tagged into epel7-testing-candidate by ellert [still active]
Tue Oct 27 13:22:30 2015: globus-gridftp-server-9.0-2.el7 tagged into epel7-override by bodhi [still active]

$ koji list-tag-history --build globus-common-16.0-1.el7
Tue Oct 27 09:13:02 2015: globus-common-16.0-1.el7 tagged into epel7-testing-candidate by ellert [still active]
Tue Oct 27 09:59:02 2015: globus-common-16.0-1.el7 tagged into epel7-override by bodhi [still active]
```
Trying to start over and make the edit again fails.

The same has happened to other updates where the same changes for other releases were attempted:

- https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2015-783ae1c4ae
  screwed up in the same way
- https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2015-e63795f6f6
  screwed up in mostly the same way - (has one less build)
- https://bodhi.fedoraproject.org/updates/FEDORA-2015-2a29bcf358
  screwed up in a slightly different way - was in a different state when the edit was done than the others, but screwed up nonetheless.

When bodhi and koji don't agree on the tagging state of a build, nothing works any more. Almost anything you try to do in bodhi with the update at that point tails in some way.

So it is a bit urgent to restore bodhi/koji in a consistent state. And then of course try to make sure that the state doesn't become inconsistent again.

**Comment by ellert** [comment]
I managed to create a completely new set of updates with the builds that were supposed to be added to or replace builds in these updates. This is not ideal, but better than not being able to do anything. Not sure what will happen to the old updates - hopefully they can be pushed so that does builds that are not replaced by builds in the new updates can become stable. I am not optimistic though.

**Comment by lmacken** [comment]
This issue should hopefully be resolved with Bodhi 2.1.3, which I just pushed to production. https://lists.fedoraproject.org/archives/list/infrastructure%40lists.fedoraproject.org/thread/WJLJGYQYGA5FZJ353GFNNSWLIK5VGWUO/
